File: EMS_BACKEND1/src/main/java/com/ems/service/DocumentService.java
package com.ems.service;

import com.ems.dto.DocumentDTO;
import org.springframework.core.io.Resource;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface DocumentService {
    DocumentDTO uploadDocument(Long employeeId, DocumentDTO documentDTO, MultipartFile file);
    DocumentDTO getDocumentById(Long documentId);
    List<DocumentDTO> getAllDocuments();
    List<DocumentDTO> getDocumentsByEmployeeId(Long employeeId);
    List<DocumentDTO> getDocumentsByDocumentType(String documentType);
    Resource downloadDocument(Long documentId);
    void deleteDocument(Long documentId);
}

File: employee-management-frontend/src/services/documentService.js
// src/services/documentService.js
import axiosInstance from "../api/axiosConfig";

const DOCUMENT_API_URL = "/documents";

const DocumentService = {
  uploadDocument: (employeeId, documentType, file) => {
    const formData = new FormData();
    formData.append("file", file);
    formData.append("documentType", documentType);
    // Optionally append fileName if you want to provide a custom one
    // formData.append("fileName", file.name);

    return axiosInstance.post(`${DOCUMENT_API_URL}/upload/${employeeId}`, formData, {
      headers: {
        "Content-Type": "multipart/form-data",
      },
    });
  },

  getDocumentById: (id) => {
    return axiosInstance.get(`${DOCUMENT_API_URL}/${id}`);
  },

  getAllDocuments: () => {
    return axiosInstance.get(DOCUMENT_API_URL);
  },

  getDocumentsByEmployeeId: (employeeId) => {
    return axiosInstance.get(`${DOCUMENT_API_URL}/employee/${employeeId}`);
  },

  getDocumentsByDocumentType: (documentType) => {
    return axiosInstance.get(`${DOCUMENT_API_URL}/type/${documentType}`);
  },

  downloadDocument: (id) => {
    return axiosInstance.get(`${DOCUMENT_API_URL}/download/${id}`, {
      responseType: "blob", // Important for downloading files
    });
  },

  deleteDocument: (id) => {
    return axiosInstance.delete(`${DOCUMENT_API_URL}/${id}`);
  },
};

export default DocumentService;
